ddsrecorder/src/cpp/tool/DdsRecorder.cpp

```diff
@@ -94,9 +93,12 @@ DdsRecorder::DdsRecorder(
         discovery_database_,
         mcap_handler_);
 
-    // Create and populate Participant Database
-    participants_database_ =
-            std::make_shared<ParticipantsDatabase>();
+    // Create an internal topic to transmit the dynamic types
+    configuration.ddspipe_configuration.builtin_topics.insert(
+        utils::Heritable<DistributedTopic>::make_heritable(type_object_topic()));
```

Comment:
NIT: I think this is the right place to have the insertion, but on the other hand I find it a bit misleading that the type object topic gets inserted to allowlist in the YAML reader (just setting the debate).

Reply:
I agree. I think it would make more sense to do both things in the DdsRecorder's constructor.